tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -3,23 +3,46 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Numerics;
 using Datadog.Trace.Logging;
+using Datadog.Trace.RemoteConfigurationManagement.Protocol;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Serilog.Events;
 
 namespace Datadog.Trace.RemoteConfigurationManagement;
 
 internal class RcmSubscriptionManager : IRcmSubscriptionManager
 {
+    private const int RootVersion = 1;
+
     public static readonly IRcmSubscriptionManager Instance = new RcmSubscriptionManager();
     private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<RcmSubscriptionManager>();
 
     private readonly List<ISubscription> _subscriptions = new();
     private readonly object _syncRoot = new();
+
+    /// <summary>
+    /// Key is the path
+    /// </summary>
+    private readonly Dictionary<string, RemoteConfigurationCache> _appliedConfigurations = new();
+
+    private readonly string _id;
+
+    private string? _backendClientState;
+    private int _targetsVersion;
     private BigInteger _capabilities;
 
+    public RcmSubscriptionManager()
```

Comment:
Forgot that. Should we make this .ctor private as the class is a singleton?

Reply:
We could, but I don't really see the point. I don't think anybody will create one by mistake, and it's nice to leave it open in case one of the tests needs it

Reply:
I'm a big fan of consistency, and most of our singleton classes have private `.ctor` with static internal `Create` method.
It doesn't make much difference, but may be we should stick to this pattern everywhere?